parquet/src/main/java/org/apache/iceberg/parquet/VectorizedParquetReader.java

```diff
@@ -49,6 +49,7 @@ public class VectorizedParquetReader<T> extends CloseableGroup implements Closea
   private final boolean caseSensitive;
   private final int batchSize;
   private final NameMapping nameMapping;
+  private int pushedLimit = -1;
 
   public VectorizedParquetReader(
```

Comment:
I think we should probably just add another constructor so we don't need to rewrite all the other use cases. I think we should also have pushedLimit just be an Integer so we can pass null. I don't like magic numbers (even -1).

So two suggestions for me
```java
// Keep
Foo (x,y,z)
// Add
Foo (x,y,z,pushedLimit)
```

And 
```java
Integer pushedLimit = null
```

Reply:
I have changed `int pushedLimit` to `Integer pushedLimit` in all the classes that use `pushedLimit`. I didn't add another constructor in `VectorizedParquetReader` because there is only one place the `VectorizedParquetReader` constructor gets called, but I have added additional constructors in other classes such as `BaseRowReader`, so we will rewrite less use cases.